NitroxReloader/Reloader.cs

```diff
@@ -0,0 +1,244 @@
+                .ToDictionary(method =>
+                {
+                    var key = QualifiedName(method);
+                    Console.WriteLine("Reloader: found reloadable method " + key);
+                    return key;
+                }, null);
+            }
+
+            public void Reload(string newAssemblyLocation)
+            {
+                var newAssembly = Assembly.Load(File.ReadAllBytes(newAssemblyLocation));
+                foreach (var method in GetReloadableMethods(newAssembly))
+                {
+                    var key = QualifiedName(method);
+                    Console.WriteLine("Reloader: patching " + key);
+                    var originalMethod = reloadableMethods[key];
+                    if (originalMethod != null)
+                    {
+                        var originalCodeStart = GetMethodStart(originalMethod);
+                        var newCodeStart = GetMethodStart(method);
+                        if (originalCodeStart == newCodeStart)
+                            Console.WriteLine("Reloader: Methods are identical! (Not emitting jump, that causes an infinite loop)");
+                        else
+                            WriteJump(originalCodeStart, newCodeStart);
+                    }
+                    else
+                        Console.WriteLine("Reloader: Original method missing (did you add a method with ReloadableMethod attribute?)");
+                }
+            }
+
+            public bool HasReloadableMethods
+            {
+                get { return reloadableMethods.Count > 0; }
+            }
+
+            public string AssemblyName
+            {
+                get { return Path.GetFileName(assembly.Location); }
+            }
+
+            #region Utils
+            private static IEnumerable<MethodInfo> GetReloadableMethods(Assembly a)
+            {
+                return a.GetTypes()
+                    .SelectMany(type => type.GetMethods(allBindings))
+                    .Where(IsMarkedReloadable);
+            }
+            private static string QualifiedName(MethodBase method)
+            {
+                return method.DeclaringType.FullName + '.' + method.Name;
+            }
+
+            private static bool IsMarkedReloadable(MethodBase method)
+            {
+                return method.GetAttribute<ReloadableMethod>() != null;
+            }
+
+            private static IntPtr GetMethodStart(MethodBase method)
+            {
+                var handle = GetRuntimeMethodHandle(method);
+                RuntimeHelpers.PrepareMethod(handle);
+                return handle.GetFunctionPointer();
+            }
+
+            public static unsafe void WriteJump(IntPtr original, IntPtr destination)
+            {
+                Console.WriteLine("Reloader: Writing jump to {0} at {1}", destination, original);
+                bool x64 = IntPtr.Size == sizeof(long);
+#if true
```

Comment:
It was to quickly toggle between the two methods of writing the jump, but I couldn't figure out why the second one doesn't do what it's supposed to do so I removed it.